rfcs/0094-use-matrix-for-official-chat.md

```diff
@@ -0,0 +1,62 @@
+---
+feature: use-matrix-for-official-chat
+start-date: 2020-05-19
+author: Ryan Mulligan
+shepherd-team:
+shepherd-leader:
+---
+
+# Summary
+[summary]: #summary
+
+Switch from IRC ([Freenode](https://freenode.net/)) to
+[Matrix](https://matrix.org/)
+([#community:nixos.org][1]) as
+our official community chat.
+
+# Motivation
+[motivation]: #motivation
+
+The current official community chat host, Freenode, has lost its
+leadership and its future is uncertain. Many Free Software communities
+are abandoning it, so its network effects may be negative instead of
+positive.
+
+There are currently 1796 members of the Unofficial NixOS Discord
+channel (244 weekly visitors, 95 weekly communicators). Some of them
+might be convinced by the more modern features available on Matrix to
+move their conversations there.
+
+# Detailed design
+[design]: #detailed-design
+
+Update the [NixOS.org community
+page](https://nixos.org/community/index.html) to point to our Matrix
+Space ([#community:nixos.org][1]) that has already been created which you
+can join by visting the [invite
+link][1].
+
+Update any other places that point to Freenode to point to Matrix
+instead.
+
+Port any bots that might be necessary such as:
+
+* nix-channel-monitor
+* github PR events
+* karma bot
+* eval bot
+* factoids bot
+
```

Comment:
On FreeNode, anyone can create new channels how they see fit. This allowed informal groups to be created without having to ask for permission.

Is it possible to create arbitrary channels on the new server? If not, what is the process to handle that?

Reply:
In principle, rooms (channels) are not tied to any server in matrix. They can have one or multiple aliases though for easier discovery and addressing, like `#nix:nixos.org` is one.
These aliases can be added later as well, allowing groups to form, grow under a non-official alias like `#nix-toasters:hardwarehackers.town` until they ask an admin of the nixos.org matrix server to give them an alias for that server as well, and add them to the space.

It might make sense to explicitly state this in the RFC and lay out a process, though.

Reply:
Rooms on Matrix do not exist "on a server" - they are replicated to all servers which have participants in the room. Anyone is free to create a room, and any amount of addresses on any homeserver can be pointed at it, and it will exist independently of the availability of any given homeserver.

I believe the default permissions required to 'claim' an alias on a homeserver (such as `#somename:matrix.org`) are 1) has admin access to the room, and 2) is using an account on the homeserver in question.